### DISCUSSION TITLE
App declined from app store because "Buttons are not responsive on tap"

### DISCUSSION THREAD

**Discussion by bartlomein** [callstack/react-native-visionos#132] [General]
Hi all, 
I understand this might be a shot in the dark, however figured I'd post. I submitted an app to review by apple and got denied because
`...During our review, we found that your app search button was unresponsive to tap.`

I have been developing this on the simulator, and all the buttons work on click. 
I have imported `import {TouchableOpacity} from '@callstack/react-native-visionos'` for all my buttons.

Wondering if anyone has ran into this issue yet while developing/uploading an app to review. Thanks!

**Comment by okwasniewski** [comment]
Hey, 

Thanks for opening the discussion. Can you provide more details? Are you using Hermes with Old Architecture/New Architecture?

In general, you should import components from `react-native` so `import {TouchableOpacity} from 'react-native'` is preferred. But importing from `react-native-visionos` shouldn't cause any issues. 

I have one app on the App Store (https://apps.apple.com/us/app/spatial-noise/id6477335349) built with RN visionOS (but with one of the previous versions 0.73.3) but there might be some regressions introduced that caused this.

**Reply by bartlomein** [reply]
Thanks so much for getting back to me. 
I believe I am using the new architecture. `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` is the command I use to install the pods. Did you use the old or new architecture with yours?
Ill change to `import {TouchableOpacity} from 'react-native'` to see if might change it.

**Reply by okwasniewski** [reply]
I've used the old arch as one of my dependencies is not supported on the new one - I'll investigate building Release build with new architecture. You might also try using the visionOS 1.1 simulator runtime (Apple is doing testing on real AVP using visionOS 1.1) so maybe something is wrong with visionOS 1.1. 

You can add visionOS 1.1 runtime by downloading it from Apple website (https://developer.apple.com/xcode/resources/) and running `xcrun simctl runtime add "VisionOS 1.1 Simulator Runtime.dmg"`

**Reply by okwasniewski** [reply]
Okay, the update to my app just got approved by the AppStore (built using 0.73.6) without any problems - I'll check if this issue is visible only on the new architecture.

**Reply by bartlomein** [reply]
Thanks so much for checking. Ill try to build with the old arch in a bit. I remember I was having some issues that you helped me out with with the old arch installing some pods, but maybe that is resolved now.

**Reply by bartlomein** [reply]
Tried to install using `bundle exec pod install` (old arch) and getting errors while running on Xcode. 
![Screenshot 2024-03-19 at 9 57 40 AM](https://github.com/callstack/react-native-visionos/assets/3741391/7dfffe69-4e70-4ec6-9819-2c337f5c0aae)

**Reply by okwasniewski** [reply]
You should first clean the build folder / delete pods `git clean -xdf visionos` or `rm -rf visionos/Pods visionos/
build`. You can also clean Xcode build folder (cmd +K) and clean Xcode derived data.

**Reply by bartlomein** [reply]
I updated to the newest visionos package, rebuilt with standard arch and resubmitted and go approved! Thanks so much for your help @okwasniewski ! Dziekuje! ;)
